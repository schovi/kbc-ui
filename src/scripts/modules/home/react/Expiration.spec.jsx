import React from 'react';
import Expiration from './Expiration';

describe('<Expiration />', function() {
  it('should render nothing when expiration is in long time', function() {
    const expires = new Date(Date.now() + (60 * 24 * 60 * 60 * 1000)).toISOString(); // in 60 days
    shallowSnapshot(<Expiration expires={expires} />);
  });

  it('should render expiration warning when expiration is close', function() {
    const expires = new Date(Date.now() + (24 * 60 * 60 * 1000)).toISOString(); // next day
    shallowSnapshot(<Expiration expires={expires} />);
  });
});