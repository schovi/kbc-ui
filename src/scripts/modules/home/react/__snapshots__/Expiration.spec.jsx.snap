exports[`<Expiration /> should render expiration warning when expiration is close 1`] = `
<div
  className="row kbc-header kbc-expiration">
  <div
    className="alert alert-warning">
    <h3>
      <span
        className="fa fa-exclamation-triangle" />
      This project will expire in 
      1 day
      .
    </h3>
    <p>
      Please 
      <a
        onClick={[Function]}>
        contact support
      </a>
       for project plan upgrade.
    </p>
  </div>
</div>
`;

exports[`<Expiration /> should render nothing when expiration is in long time 1`] = `null`;
